src/decoder/image.rs

```diff
@@ -624,19 +624,25 @@ impl Image {
             self.jpeg_tables.as_deref().map(|a| &**a),
         )?;
 
+        // Polyfil for usize::div_ceil added in rust 1.73+
+        fn usize_div_ceil(numerator: usize, denominator: usize) -> usize {
+            (numerator + denominator - 1) / denominator
+        }
+
```

Comment:
Could you add a comment here explaining what a "buffer unit" is? Though a shorter name might be better. Maybe call them "elements"?